packages/taquito-michelson-encoder/test/viewsSchema.spec.ts

```diff
@@ -1,117 +1,157 @@
+  });
 
+  it('Should throw if view does not have the right length of elements', () => {
+    expect(
+      () => new ViewSchema([{ string: 'add' }, { prim: 'nat' }, { prim: 'nat' }])
+    ).toThrowError(InvalidScriptError);
+  });
 });
 
-
 describe('ViewSchema.fromRPCResponse test', () => {
+  it('Should create a ViewSchema instance for each view in the contract code', () => {
+    const viewSchemas = ViewSchema.fromRPCResponse({ script });
+    expect(viewSchemas.length).toEqual(7);
+    expect(viewSchemas[0].viewName).toEqual('add');
+    expect(viewSchemas[1].viewName).toEqual('fib');
+    expect(viewSchemas[2].viewName).toEqual('id');
+    expect(viewSchemas[3].viewName).toEqual('is_twenty');
+    expect(viewSchemas[4].viewName).toEqual('step_constants');
+    expect(viewSchemas[5].viewName).toEqual('succ');
+    expect(viewSchemas[6].viewName).toEqual('test_failwith');
+  });
 
-    it('Should create a ViewSchema instance for each view in the contract code', () => {
-        const viewSchemas = ViewSchema.fromRPCResponse({ script });
-        expect(viewSchemas.length).toEqual(7);
-        expect(viewSchemas[0].viewName).toEqual('add');
-        expect(viewSchemas[1].viewName).toEqual('fib');
-        expect(viewSchemas[2].viewName).toEqual('id');
-        expect(viewSchemas[3].viewName).toEqual('is_twenty');
-        expect(viewSchemas[4].viewName).toEqual('step_constants');
-        expect(viewSchemas[5].viewName).toEqual('succ');
-        expect(viewSchemas[6].viewName).toEqual('test_failwith');
-    });
+  it('Should return an empty array if there is no view in the contract code', () => {
+    const viewSchemas = ViewSchema.fromRPCResponse({ script: rpcContractResponse.script });
+    expect(viewSchemas.length).toEqual(0);
+  });
 
-    it('Should return an empty array if there is no view in the contract code', () => {
-        const viewSchemas = ViewSchema.fromRPCResponse({ script: rpcContractResponse.script });
-        expect(viewSchemas.length).toEqual(0);
-    });
+  it('Should throw InvalidScriptError if view does not have the right length of elements', () => {
+    expect(() =>
+      ViewSchema.fromRPCResponse({
+        script: {
+          code: [
+            {
+              prim: 'view',
+              args: [{ string: 'add' }, { prim: 'nat' }, { prim: 'nat' }],
+            },
+          ],
+          storage: {},
+        },
+      })
+    ).toThrowError(InvalidScriptError);
+  });
 
-    it('Should throw if view does not have the right length of elements', () => {
-        expect(() =>
-            ViewSchema.fromRPCResponse({
-                script: {
-                    code: [{
-                        prim: 'view',
-                        args: [{ string: 'add' }, { prim: 'nat' }, { prim: 'nat' }]
-                    }], storage: {}
-                }
-            })).toThrowError(InvalidScriptError);
+  it('Should throw InvalidRpcResponseError if storage is not an array', () => {
```

Comment:
This file was for testing the class `ViewSchema`. It might not be best place for the tests `Should throw InvalidRpcResponseError if storage is not an array` and `Should throw InvalidBigMapSchema if big map schema is invalid'`. Consider moving them to another file `Schema` maybe

Reply:
Created a new file schema.spec.ts and moved the two tests to it.